### DISCUSSION TITLE
Row selection not working for nested rows (getSubRows)

### DISCUSSION THREAD

**Discussion by srikrishna212** [KevinVandy/material-react-table#684] [General]
**Describe the bug**
row-selection in the group rows are not correctly reacting to their sub-rows row-selection state

**Steps to reproduce**

Expand the first group row
Select the first sub-row
✅ Notice the parent row checkbox being "indeterminate"

Select the other 2 sub-rows in that same group
🛑 Notice the parent row checkbox is "unchecked", when it should be "checked"

**Expected behavior**
As a user, I expected the parent row to react to the row-selection status of its children, regardless on how they are selected/unselected:

some sub-rows selected: parent row "indeterminate"
no sub-rows selected: parent row "unchecked"
all sub-rows selected: parent row "checked"

**Comment by sund168** [comment]
Yes, I met the same issue. 

Here's the example at the official site: https://www.material-react-table.dev/?path=/story/features-sub-row-tree-examples--sub-row-tree-with-selection

Expand the group row... when all the sub rows are selected, the checkbox in the group row doesn't respond accordingly.
